File: type-script-nest-graph-ql-실습/src/app/software/model/dto/response/software-response.dto.ts
import { SoftwareEntity } from "../../entities/software.entity";

export class SoftwareResponseDto {
  private softwareId: string;
  private usedBy: string;
  private developedBy: string;
  private description: string;

  constructor(entity: SoftwareEntity) {
    this.softwareId = entity.softwareId;
    this.usedBy = entity.usedBy;
    this.developedBy = entity.developedBy;
    this.description = entity.description;
  }
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/api/software/model/dto/response/SoftwareResponseDto.kt
package com.junyharangstudy.kotlingraphqltest.api.software.model.dto.response

import com.junyharangstudy.kotlingraphqltest.api.software.model.entity.Software

class SoftwareResponseDto (
    private val softwareId: String,
    private val usedBy: String,
    private val developedBy: String,
    private val description: String?
) {
    companion object {
        fun softwareToDto(software: Software): SoftwareResponseDto {
            return SoftwareResponseDto(
                softwareId = software.softwareId,
                usedBy = software.usedBy,
                developedBy = software.developedBy,
                description = software.description)
        }
    }
}